File: src/main/java/com/autosale/shop/model/PaginationResponse.java
package com.autosale.shop.model;

import lombok.Value;

import java.util.List;

@Value
public class PaginationResponse<T> {
    List<T> content;
    int pages;
    int currentPage;
    int pageSize;

    public PaginationResponse(List<T> content, PaginationRequest request, int itemsAmount) {
        this.content = content;
        pages = (int) Math.ceil((double) itemsAmount / request.getPageSize());
        currentPage = request.getCurrentPage();
        pageSize = request.getPageSize();
    }
}

File: src/main/java/com/autosale/shop/model/PaginationResponse.kt
package com.autosale.shop.model

import kotlin.math.ceil


data class PaginationResponse<T>(
    val content: List<T>,
    val pages: Int,
    val currentPage: Int,
    val pageSize: Int
) {
    constructor(content: List<T>, request: PaginationRequest, itemsAmount: Int) :
            this(
                content,
                ceil(itemsAmount.toDouble() / request.pageSize).toInt(),
                request.currentPage,
                request.pageSize
            )
}



